src/server/search/search_family.cc

```diff
@@ -140,48 +154,51 @@ optional<search::Schema> ParseSchemaOrReply(DocIndex::DataType type, CmdArgParse
 }
 
 optional<SearchParams> ParseSearchParamsOrReply(CmdArgParser parser, ConnectionContext* cntx) {
-  size_t limit_offset = 0, limit_total = 10;
-
-  optional<SearchParams::FieldReturnList> return_list;
-  search::QueryParams query_params;
+  SearchParams params;
 
   while (parser.ToUpper().HasNext()) {
     // [LIMIT offset total]
     if (parser.Check("LIMIT").ExpectTail(2)) {
-      limit_offset = parser.Next().Int<size_t>();
-      limit_total = parser.Next().Int<size_t>();
+      params.limit_offset = parser.Next().Int<size_t>();
+      params.limit_total = parser.Next().Int<size_t>();
       continue;
     }
 
     // RETURN {num} [{ident} AS {name}...]
     if (parser.Check("RETURN").ExpectTail(1)) {
       size_t num_fields = parser.Next().Int<size_t>();
-      return_list = SearchParams::FieldReturnList{};
-      while (return_list->size() < num_fields) {
+      params.return_fields = SearchParams::FieldReturnList{};
+      while (params.return_fields->size() < num_fields) {
         string_view ident = parser.Next();
         string_view alias = parser.Check("AS").IgnoreCase().ExpectTail(1) ? parser.Next() : ident;
-        return_list->emplace_back(ident, alias);
+        params.return_fields->emplace_back(ident, alias);
       }
       continue;
     }
 
     // NOCONTENT
     if (parser.Check("NOCONTENT")) {
-      return_list = SearchParams::FieldReturnList{};
+      params.return_fields = SearchParams::FieldReturnList{};
       continue;
     }
 
     // [PARAMS num(ignored) name(ignored) knn_vector]
     if (parser.Check("PARAMS").ExpectTail(1)) {
       size_t num_args = parser.Next().Int<size_t>();
-      while (parser.HasNext() && query_params.Size() * 2 < num_args) {
+      while (parser.HasNext() && params.query_params.Size() * 2 < num_args) {
         string_view k = parser.Next();
         string_view v = parser.Next();
-        query_params[k] = v;
+        params.query_params[k] = v;
       }
       continue;
     }
 
+    if (parser.Check("SORTBY").ExpectTail(1)) {
+      params.sort_option =
+          search::SortOption{string{parser.Next()}, bool(parser.Check("DESC").IgnoreCase())};
```

Comment:
Does this mean that `DESC` is expected to directly follow `SORTBY`?
(see my above comment)

Reply:
No, you have a parser.Next() before it